### ISSUE TITLE
Install the letsencrypt package when available

### ISSUE THREAD

**Issue by thefinn93** [thefinn93/ansible-letsencrypt#10]
Installing the package is far less janky than compiling from source every time. Looks like this is available in [Ubuntu Xenial](http://packages.ubuntu.com/xenial/letsencrypt) and [Debian Stretch](https://packages.debian.org/stretch/letsencrypt)

**Comment by jaywink** [comment]
Thanks for this! Tried and works at least on 15.10 (wily) too, by downloading `letsencrypt`, `python-letsencrypt` and `python-acme` debs - the rest of the deps are found.

**Comment by thefinn93** [comment]
I'd rather not download `.deb`s targeted at other versions. If it's not in the repos, we should probably just build it as we do now.

**Comment by pmackay** [comment]
@thefinn93 what about installing the packages on those distros that have them?

**Comment by thefinn93** [comment]
@pmackay yeah, that's what the point of this issue is. "When available" i guess was my way of saying that. When it's in the package manager.

**Comment by dothebart** [comment]
```

- name: add hub-binary for github remoting
  apt_repository: repo='ppa:certbot/certbot'

- name: update apt cache
  apt: update_cache=yes cache_valid_time=3600

- name: install certbot
  apt: name=certbot
```
should work recently as seen on https://github.com/certbot/certbot/issues/1706

**Comment by thefinn93** [comment]
@dothebart does that work on Debian? I was under the impression that PPA's were sort of an Ubuntu thing.

**Comment by thefinn93** [comment]
also, looks like it doesn't detect the availability of the package so much as add a repo and hope it works

**Comment by rfleschenberg** [comment]
Installing with pip is not supported by certbot. The current PyPI package is broken. The certbot developers recommend to install from backports on Debian Jessie: https://github.com/certbot/certbot/issues/5015

So, it seems that installing from distribution packages whenever possible is indeed the way to go.

**Comment by rfleschenberg** [comment]
I don't want to promise anything, but I will try to find the time to start a PR this weekend.

**Comment by ArcolaJasper** [comment]
I need this for work now, so I'm working on it. will be Debian only unfortunatly :crying_cat_face:

**Comment by ArcolaJasper** [comment]
PR, needs testing on non-Debian hosts: https://github.com/thefinn93/ansible-letsencrypt/pull/59
